pennylane/ops/op_math/controlled.py

```diff
@@ -866,3 +878,25 @@ def parameter_frequencies(self):
             "and parameter frequencies can not be computed via generator for more than one "
             "parameter."
         )
+
+
+# Program capture with controlled ops needs to unpack and re-pack the control wires to support dynamic wires
+# See capture module for more information on primitives
+# If None, jax isn't installed so the class never got a primitive.
+if Controlled._primitive is not None:  # pylint: disable=protected-access
+
+    @Controlled._primitive.def_impl  # pylint: disable=protected-access
+    def _(base, *control_wires, control_values=None, work_wires=None, id=None):
+        return type.__call__(
+            Controlled,
+            base,
+            control_wires,
+            control_values=control_values,
+            work_wires=work_wires,
+            id=id,
+        )
+
+
+# easier to just keep the same primitive for both versions
+# dispatch between the two types happens inside instance creation anyway
+ControlledOp._primitive = Controlled._primitive  # pylint: disable=protected-access
```

Comment:
Funky! I like this reduction, but we should be aware that this breaks the "an operator's primitive has the operator's name as name standard"

Reply:
Basically, `ControlledOp` primitives was never actually getting used, because we always relied on `Controlled.__new__` to defer to it. We don't really need to distinguish between the two at the capture level.